packages/core/src/components/Form.tsx

```diff
@@ -575,12 +585,35 @@ export default class Form<
   omitExtraData = (formData?: T): T | undefined => {
     const { schema, schemaUtils } = this.state;
     const retrievedSchema = schemaUtils.retrieveSchema(schema, formData);
+
     const pathSchema = schemaUtils.toPathSchema(retrievedSchema, '', formData);
     const fieldNames = this.getFieldNames(pathSchema, formData);
     const newFormData = this.getUsedFormData(formData, fieldNames);
     return newFormData;
   };
 
+  // Filtering errors based on your retrieved schema to only show errors for properties in the selected branch.
+  private filterErrorsBasedOnSchema(schemaErrors: ErrorSchema<T>, resolvedSchema?: S, formData?: any): ErrorSchema<T> {
+    const { retrievedSchema, schemaUtils } = this.state;
+    const _retrievedSchema = resolvedSchema ?? retrievedSchema;
+    const pathSchema = schemaUtils.toPathSchema(_retrievedSchema, '', formData);
+    const fieldNames = this.getFieldNames(pathSchema, formData);
+    const filteredErrors: ErrorSchema<T> = _pick(schemaErrors, fieldNames as unknown as string[]);
+    // Removing undefined and empty errors.
+    const filterUndefinedErrors = (errors: any): ErrorSchema<T> => {
+      Object.keys(errors).forEach((key: string) => {
+        const errorKey = key as keyof typeof errors;
+        if (errors[errorKey] === undefined) {
+          delete errors[errorKey];
+        } else if (typeof errors[errorKey] === 'object' && !Array.isArray(errors[errorKey].__errors)) {
+          filterUndefinedErrors(errors[errorKey]);
+        }
```

Comment:
Use a local variable to reduce the operations slightly
```suggestion
        const errorAtKey = errors[errorKey];
        if (errorAtKey === undefined) {
          delete errors[errorKey];
        } else if (typeof errorAtKey === 'object' && !Array.isArray(errorAtKey.__errors)) {
          filterUndefinedErrors(errorAtKey);
        }
```